### PR TITLE
pass environment name to managers and add documentation for expected …

### PR DESCRIPTION
**Pull request by KPostOffice**
## Related Issues and Dependencies

#715 
This is laying the ground work for solving the above issue. It allows for passing an environment to Kebechet as well as outlining the desired behaviour of managers which handle overlays.

Please take a look at the desired behaviour I defined in the managers/README.rst and feel free to suggest changes.

## This introduces a breaking change

- [ ] Yes
- [x] No

### CODE REVIEW

File: kebechet/managers/README.rst [hunk 1]
```diff
@@ -36,5 +36,22 @@ If you wish to operate on repository source code, you can request to clone it:
 
 The last thing you need to do, is to register your manager to `REGISTERED_MANAGERS` constant (you can find it in `kebechet/managers/__init__.py` file) so Kebechet knows about your manager. Manager can be referenced by its name in lowercase (class name without the "manager" suffix).
 
+Overlays
+========
+
+Thoth allows users to specify overlays consisting of different hardware and software environments.  These environments
```
Comment (pacospace) [inline kebechet/managers/README.rst:42]: can we call them runtime environment to be consistent?

File: kebechet/managers/README.rst [hunk 2]
```diff
@@ -36,5 +36,22 @@ If you wish to operate on repository source code, you can request to clone it:
...
+are specified in a users .thoth.yaml file, files associated with a specific overlay are located in
+`<overlays-dir>/<environment-name>`.  Thoth will specify overlays to act on, if you create a manager which acts on a
```
Comment (pacospace) [inline kebechet/managers/README.rst:44]: what do you mean by a manager acting on a specific overlay? Isn't the default behaviour, act on all overlays if `overlays_dir` key exists in .thoth.yaml?
In that case go over all runtime enviornment names in `runtime_environments`, check for overlays/runtime_environment_name and act on those opening PR with updated dependencies, if no match is found, open an issue saying you have this overlay but .thoth.yaml please be sure to use same name

If non`overlays_dir` key exists default to usual behaviour, so look for dependencies in the root and use the first runtime environment in the list in .thoth.yaml by default

File: kebechet/managers/README.rst [hunk 3]
```diff
@@ -36,5 +36,22 @@ If you wish to operate on repository source code, you can request to clone it:
...
+specific overlay then the desired behaviour is as follows.
+
+* if no overlays directory is specified in .thoth.yaml and no environment is passed, then the manager should act only on
+the first entry in `environments` in .thoth.yaml and changes should be made to the top level directory
```
Comment (pacospace) [inline kebechet/managers/README.rst:48]: `runtime_environments` so we are consisten with nomenclature

File: kebechet/managers/README.rst [hunk 4]
```diff
@@ -36,5 +36,22 @@ If you wish to operate on repository source code, you can request to clone it:
...
+* if an overlays directory is specified and no environment is passed the manager should act on every overlay and make
```
Comment (pacospace) [inline kebechet/managers/README.rst:49]: what you mean no environment is passed? I think runtime_environment is mandatory key in .thoth.yaml file, so it should have at least one right? cc @fridex

File: kebechet/cli.py [hunk 5]
```diff
@@ -77,15 +77,21 @@ def cli(ctx=None, verbose=0):
 @click.option("-n", "--namespace", envvar="KEBECHET_PROJECT_NAMESPACE", required=True)
 @click.option("-p", "--project", envvar="KEBECHET_PROJECT_NAME", required=True)
 @click.option("-u", "--service-url", envvar="KEBECHET_SERVICE_URL")
+@click.option("-e", "--environment-name", envvar="KEBECHET_ENVIRONMENT_NAME")
```
Comment (pacospace) [inline kebechet/cli.py:80]: let's use consistent nomenclature, `RUNTIME_ENVIRONMENT_NAME`

**Review by pacospace — COMMENTED** [review]
Few comments, well done!!

File: kebechet/managers/README.rst (hunk 4 above)
Comment (fridex) [reply kebechet/managers/README.rst:49]: Yes, the key is mandatory, we do not require any entries in it though (can be an empty list). This is something we could improve on thamos side.

Comment (KPostOffice) [reply kebechet/managers/README.rst:49]: Here I am talking about whether `runtime_environment` was passed to `kebechet`

File: kebechet/managers/README.rst [hunk 6]
```diff
@@ -36,5 +36,23 @@ If you wish to operate on repository source code, you can request to clone it:
...
+Overlays
+========
+
+Thoth allows users to specify overlays consisting of different runtime environments.  These environments are specified
+in a users .thoth.yaml file, files associated with a specific runtime environment are located in
+`<overlays-dir>/<runtime-environment-name>`.  If you create a manager which acts on individual runtime environments,
+then the desired behaviour is as follows.
+
+* if no overlays directory is specified in .thoth.yaml and no runtime environment is passed, then the manager should act
+only on the first entry in `environments` in .thoth.yaml and changes should be made to the top level directory
```
Comment (pacospace) [inline kebechet/managers/README.rst:48]: the entry is `runtime_environments`

File: kebechet/managers/README.rst [hunk 7]
```diff
@@ -36,5 +36,23 @@ If you wish to operate on repository source code, you can request to clone it:
...
+* if an overlays directory is specified and no environment is passed the manager should act on every runtime environment
```
Comment (pacospace) [inline kebechet/managers/README.rst:49]: is specified in .thoth.yaml?

File: kebechet/managers/README.rst [hunk 8]
```diff
@@ -36,5 +36,23 @@ If you wish to operate on repository source code, you can request to clone it:
 
 The last thing you need to do, is to register your manager to `REGISTERED_MANAGERS` constant (you can find it in `kebechet/managers/__init__.py` file) so Kebechet knows about your manager. Manager can be referenced by its name in lowercase (class name without the "manager" suffix).
 
+Overlays
+========
+
+Thoth allows users to specify overlays consisting of different runtime environments.  These environments are specified
```
Comment (pacospace) [inline kebechet/managers/README.rst:42]: These runtime environments

**Review by pacospace — APPROVED** [review]
lgtm!

**Pull request merged**